x-pack/legacy/plugins/apm/public/components/app/ServiceNodeMetrics/index.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiHorizontalRule,
+  EuiFlexGrid,
+  EuiPanel,
+  EuiSpacer,
+  EuiStat
+} from '@elastic/eui';
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import styled from 'styled-components';
+import { ApmHeader } from '../../shared/ApmHeader';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { useAgentName } from '../../../hooks/useAgentName';
+import { useServiceMetricCharts } from '../../../hooks/useServiceMetricCharts';
+import { ChartsSyncContextProvider } from '../../../context/ChartsSyncContext';
+import { MetricsChart } from '../../shared/charts/MetricsChart';
+import { useFetcher, FETCH_STATUS } from '../../../hooks/useFetcher';
+import { callApmApi } from '../../../services/rest/callApmApi';
+import { truncate, px, unit } from '../../../style/variables';
+
+const INITIAL_DATA = {
+  host: '',
+  containerId: ''
+};
+
+const Truncate = styled.span`
+  display: block;
+  ${truncate(px(unit * 12))}
+`;
+
+export function ServiceNodeMetrics() {
+  const { urlParams } = useUrlParams();
+  const { serviceName, serviceNodeName } = urlParams;
+
+  const { agentName } = useAgentName();
+  const { data } = useServiceMetricCharts(urlParams, agentName);
+  const { start, end } = urlParams;
+
+  const {
+    data: { host, containerId } = INITIAL_DATA,
+    status
+  } = useFetcher(() => {
+    if (serviceName && serviceNodeName) {
+      return callApmApi({
+        pathname:
+          '/api/apm/services/{serviceName}/node/{serviceNodeName}/metadata',
+        params: {
+          path: { serviceName, serviceNodeName }
+        }
+      });
+    }
+  }, [serviceName, serviceNodeName]);
+
+  const isLoading =
+    status === FETCH_STATUS.LOADING || status === FETCH_STATUS.PENDING;
```

Comment:
Is it necessary to take `FETCH_STATUS.PENDING` into account?

Reply:
Well, I think in this case no, because `serviceName` and `serviceNodeName` are always defined so it should never be `FETCH_STATUS.PENDING`. But I think that would easily break. Maybe we should expose `hasNotYetLoaded` (or `hasLoadedOnce`) from within the `useFetcher` hook?

Reply:
`hasNotYetLoaded` doesn't feel like something generic that fits in a http requester. After the fact I checked how other models http states and turns out we are 1:1 aligned with Elm (apart from naming):

https://elmprogramming.com/remote-data.html#handling-not-asked-state

So I think the four states we have are ok. I think it's very rare we'll need the pending state.

Reply:
Removed the check for `FETCH_STATUS.PENDING` 👍 .